operator/internal/controller/vectorized/cluster_controller_configuration.go

```diff
@@ -177,19 +179,31 @@ func (r *ClusterReconciler) applyPatchIfNeeded(
 	log := l.WithName("applyPatchIfNeeded")
 	errorWithContext := newErrorWithContext(redpandaCluster.Namespace, redpandaCluster.Name)
 
-	var invalidProperties []string
-	for i := range status {
-		invalidProperties = append(invalidProperties, status[i].Invalid...)
-		invalidProperties = append(invalidProperties, status[i].Unknown...)
-	}
-
-	patch := configuration.ThreeWayMerge(log, cfg.ClusterConfiguration, clusterConfig, lastAppliedConfiguration, invalidProperties, schema)
-	if patch.Empty() {
-		return true, nil
-	}
+	// Massage the clusterConfig into an appropriate set of values.
+	// Because we're going to perform a declarative application, invalid values will either
+	// be overwritten (if they're supplied in the cfg), or removed (if they're not).
+	// No additional handling is needed.
+	properties := map[string]any{}
+	var keys []string
+	for k, v := range cfg.ClusterConfiguration {
+		metadata := schema[k]
+		properties[k] = configuration.ParseConfigValueBeforeUpsert(log, v, &metadata)
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+
+	// Unconditionally apply the update
+	syncer := syncclusterconfig.Syncer{
+		Client: adminAPI,
+		Mode:   syncclusterconfig.SyncerModeDeclarative,
+		EqualityCheck: func(key string, desired, current any) bool {
+			return configuration.PropertiesEqual(log, desired, current, schema[key])
+		},
+	}
+	log.Info("Applying the cluster configuration", "cfg", strings.Join(keys, " "))
```

Comment:
nit: Could we push this log line into the Syncer struct?

Reply:
done